apps/condo/domains/billing/schema/RegisterBillingReceiptsService.js

```diff
@@ -117,10 +122,22 @@ const RegisterBillingReceiptsService = new GQLCustomSchema('RegisterBillingRecei
 
                 // preserve order for response
                 // errors are critical and receipt will be skipped, problems can be fixed later
-                let receiptIndex = Object.fromEntries(receiptsInput.map((receiptInput, index) =>
-                    ([index, { ...receiptInput, error: null, problems: [] }])
-                ))
+                let receiptIndex = {}
                 let errorsIndex = {}
+                let unchangedIndex = {}
+                const cache = new ReceiptInputCache(redisClient, getRedisKey, billingContextId)
+
+                for (let index = 0; index < receiptsInput.length; index++) {
```

Comment:
it looks like some internal cache logic...

When adding caching, I would expect to see a simpler structure in the code, like:

```javascript
const cache = new XXX(…);
…
if (cache.isChanged(receipt)) {
 …
}
```

However, it seems there are quite a few changes here. Perhaps we should reconsider the approach to code organization?